framework/src/Volo.Abp.IdentityModel/Volo/Abp/IdentityModel/IdentityModelTokenCacheItem.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using System.Linq;
+using IdentityModel.Client;
+using Volo.Abp.MultiTenancy;
+
+namespace Volo.Abp.IdentityModel
+{
+    [Serializable]
+    [IgnoreMultiTenancy]
+    public class IdentityModelTokenCacheItem
+    {
+        public string AccessToken { get; set; }
+
+        public IdentityModelTokenCacheItem()
+        {
+
+        }
+
+        public IdentityModelTokenCacheItem(string accessToken)
+        {
+            AccessToken = accessToken;
+        }
+
+        public static string CalculateCacheKey(DiscoveryDocumentResponse discoveryResponse, IdentityClientConfiguration configuration)
+        {
+            return discoveryResponse.TokenEndpoint + string.Join(",", configuration.Select(x => x.Key + ":" + x.Value));
```

Comment:
Instead of such a long key (which may have problems with cache providers), can we get MD5 (or use a better algorithm) hash of the string. I am just thiking and not sure about it. What do you think?

Reply:
I have considered md5, but the name of the key will be meaningless.

Reply:
If the name of the key does not need to have any meaning, **md5** is no problem.